bin/transform-genbank

```diff
@@ -0,0 +1,280 @@
+    name structure and print them out for manual curation. Drop the problem
+    records and duplicate records and return the modified DataFrame.
+    """
+    strain_name_regex = r'([\w]*/)?[\w]*/[-_\.\w]*/[\d]{4}'
+
+    problem_data = genbank_data.loc[(genbank_data.region == '') \
+        | (genbank_data.country == '') \
+        # All strain names should have structure {}/{}/{year} or {}/{}/{}/{year}
+        # with the exception of 'Wuhan-Hu-1/2019'
+        | (~(genbank_data.strain.str.match(strain_name_regex)) & (genbank_data.strain != 'Wuhan-Hu-1/2019'))]
+
+    # Print problem records for manual curation
+    problem_data[['genbank_accession', 'strain', 'region', 'country', 'url']].to_csv(args.problem_data, sep='\t', index=False)
+
+    # Drop entries without geographic region or with wrong strain name structure
+    genbank_data.drop(problem_data.index, inplace=True)
+
+    # Drop duplicates, prefer longest and earliest sequence
+    # based on submission date.
+    genbank_data.sort_values(['strain', 'length', 'date_submitted'],
+        ascending=[True, False, True], inplace=True)
+    genbank_data.drop_duplicates('strain', inplace=True)
+
+    return genbank_data
+
+
+if __name__ == '__main__':
+    base = Path(__file__).resolve().parent.parent
+
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("genbank_data",
+        default="s3://nextstrain-ncov-private/genbank.ndjson.gz",
+        nargs="?",
+        help="Newline-delimited GenBank JSON data")
+    parser.add_argument("--annotations",
+        default=base / "source-data/genbank_annotations.tsv",
+        help="Optional manually curated annotations TSV.\n"
+            "The TSV file should have no header and exactly four columns which contain:\n\t"
+            "1. the GenBank accession number\n\t"
+            "2. the column name to replace from the generated `metadata.tsv` file\n\t"
+            "3. the replacement data\n"
+        "Lines or parts of lines starting with '#' are treated as comments.\n"
+        "e.g.\n\t"
+        "MT039888	location    Boston\n\t"
+        "# First Californian sample\n\t"
+        "MN994467	country_exposure	China\n\t"
+        "MN908947	collection_date 2019-12-26 # Manually corrected date")
+    parser.add_argument("--output-metadata",
+        default=base / "data/genbank/metadata.tsv",
+        help="Output location of generated metadata tsv. Defaults to `data/genbank/metadata.tsv`")
+    parser.add_argument("--output-fasta",
+        default=base / "data/genbank/sequences.fasta",
+        help="Output location of generated FASTA file. Defaults to `data/genbank/sequences.fasta`")
+    parser.add_argument("--problem-data",
+        default=base / "data/genbank/problem_data.tsv",
+        help="Output location of generated tsv of problem records missing geography region or a valid strain name")
```

Comment:
I wonder if instead of naming this file _problem_data.tsv_, we could create an _exclude.txt_ and concatenate it with the one downstream in `ncov` like #43 originally proposed.

Reply:
This is a little different from `exclude.txt`. The `problem_data.tsv` contains records of sequences that are not included in the `metadata.tsv` at all because they are missing required location fields or they have improperly formatted strain names.  So they are not even passed downstream to the ncov build at all.